server/src/storage.rs

```diff
@@ -8,10 +8,38 @@ use std::sync::Mutex;
 use tracing::instrument;
 use warp::http::StatusCode;
 
-#[derive(Default)]
-pub(crate) struct StorageEntry {
-    reports: Vec<SignedReport>,
-    bytes: Option<Vec<u8>>,
+pub(crate) enum StorageEntry {
+    /// The storage entry is accepting new reports.
+    Open(Vec<SignedReport>),
+    /// The storage entry is finalized, and contains a serialization of
+    /// all reports for the time interval in random order.
+    Sealed(Vec<u8>),
+}
+
+impl Default for StorageEntry {
+    fn default() -> Self {
+        StorageEntry::Open(Vec::default())
+    }
+}
+
+impl StorageEntry {
+    /// Seal the entry, if it is open.
+    fn seal(&mut self) {
```

Comment:
Technically could just return a vector here

Reply:
How would that work? `seal` has to flip the enum from `Open` to `Sealed`.